joplin/base/views/publish_succeeded.py

```diff
@@ -12,22 +13,40 @@
 logger = logging.getLogger('joplin')
 
 
-'''
-    Updates "publish_*" fields in base_model for pages that triggered publish requests.
-'''
 @api_view(['POST'])
 @permission_classes([HasAPIKey])
 def publish_succeeded(request):
+    '''
+        Updates "publish_*" fields in base_model for pages that triggered publish requests.
+        This is a "fire and forget" endpoint. Publisher doesn't care what happens to the page data once it reaches Joplin.
+        Page data is processed in a separate thread. The code finishes executing after Publisher gets its response.
+        Otherwise if we update more than 200 documents, this endpoint would time out.
+            Parameters:
+                request.body.pages (list of page_data)
+                request.body.api_keys (list of str)
+            Returns:
+                Response (int): always return 200 success code.
+    '''
     body = json.loads(request.body)
     pages = body["pages"]
+    api_keys = body["api_keys"]
+    threading.Thread(target=handle_publish_success, args=(pages,api_keys)).start()
```

Comment:
By creating a separate Thread, that code can keep executing even after this function finishes and returns a response to Publisher.